tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -1035,10 +1053,31 @@ class RingAllgatherOverlapTest : public MultiDeviceTest {
     auto tc_unsharded_expected_ =
         torch::matmul(ta_unsharded_, tb_unsharded_.cpu());
     EXPECT_TRUE(
-        tc_unsharded_.cpu().allclose(tc_unsharded_expected_, 1e-1, 1e-1))
+        tc_unsharded_.cpu().allclose(tc_unsharded_expected_, 5e-1, 5e-1))
```

Comment:
This can and may already hide real errors. One trick I did before was to give input matrices integer values `at::randint` so the result is less sensitive to the computation order. But I had to be careful with the sizes so things don't overflow and we don't end up with just comparing infs.